torch/csrc/cuda/memory_snapshot.h

```diff
@@ -10,7 +10,11 @@ namespace cuda {
 // those defined in cuda/Module.cpp which also record python state.
 TORCH_CUDA_CU_API void _record_memory_history(
     bool enabled,
-    int64_t alloc_trace_max_entries = 1);
+    bool record_context = true,
+    int64_t trace_alloc_max_entries = 1,
+    bool trace_alloc_record_context = false,
+    bool record_cpp_context = false);
```

Comment:
hella boolean blindness here

Reply:
It just matches what the internal Python API is today. I am going to clean it up once I stop adding features and I can bin them into a couple sane options.